### PR TITLE
adding mock hooks & null check in toolrunner

### PR DESCRIPTION
**Pull request by jasholl**
Need to expose the following methods in task so I can mock them in L0 tests in vsts-tasks

process.cwd(): string;
os.type(): string;
fs.readdirSync(path: string): string[];

Also adding a null check in tool runner so if the tool is not kicked off properly and results in no stdout or stderr, node doesn't just die but instead bubbles up a failure to the offending task.

### CODE REVIEW

File: node/lib/task.ts [hunk 1]
```diff
@@ -591,6 +591,47 @@ export function exist(path: string): boolean {
 }
 
 /**
+ * Reads a directory's contents.
+ * see [fs.readdirsync](https://nodejs.org/api/fs.html#fs_fs_readdirsync_path_options)
+ * 
+ * @param       path    path of directory to readdirSync
+ * @returns     array of directory contents
+ */
+export function readdirSync(path: string): string[] {
```
Comment (bryanmacfarlane) [inline node/lib/task.ts]: we have been wrapping shelljs with other disk operations.  That would mean we add ls function, not readdirSync.

File: node/lib/task.ts [hunk 2]
```diff
@@ -591,6 +591,47 @@ export function exist(path: string): boolean {
...
+}
+
+/**
+ * Deletes the specified file.
+ * see [fs.unlinkSync](https://nodejs.org/api/fs.html#fs_fs_unlinksync_path)
+ * 
+ * @param        path   path of the file to delete
+ */
+export function unlinkSync(path: string): void {
+    fs.unlinkSync(path);
```
Comment (bryanmacfarlane) [inline node/lib/task.ts]: shelljs rm?  We already have rmrf function.

File: node/lib/task.ts [hunk 3]
```diff
@@ -591,6 +591,47 @@ export function exist(path: string): boolean {
...
+    return os.type();
+}
+
+/**
+ * Returns the process's current working directory.
+ * see [process.cwd](https://nodejs.org/api/process.html#process_process_cwd)
+ * 
+ * @return      the path to the current working directory of the process
+ */
+export function processCwd() : string {
```
Comment (bryanmacfarlane) [inline node/lib/task.ts]: would prefer cwd method

File: node/lib/toolrunner.ts [hunk 4]
```diff
@@ -376,8 +377,8 @@ export class ToolRunner extends events.EventEmitter {
         }
 
         var res:IExecResult = <IExecResult>{ code: r.status, error: r.error };
-        res.stdout = r.stdout.toString();
-        res.stderr = r.stderr.toString();
+        res.stdout = (r.stdout) ? r.stdout.toString() : null;
+        res.stderr = (r.stderr) ? r.stderr.toString() : null;
```
Comment (bryanmacfarlane) [inline node/lib/toolrunner.ts]: ok

File: node/lib/task.ts (hunk 1 above)
Comment (jasholl) [inline node/lib/task.ts]: added ls instead.

File: node/lib/task.ts (hunk 2 above)
Comment (jasholl) [inline node/lib/task.ts]: didn't see that method before; switched to using rmRF and pulled readdirSync out.

File: node/lib/task.ts (hunk 3 above)
Comment (jasholl) [inline node/lib/task.ts]: added cwd instead as requested.

**Pull request merged**